src/junit.rs

```diff
@@ -10,36 +10,42 @@ use quick_xml::reader::Reader;
 use crate::testrun::{Outcome, Testrun};
 
 // from https://gist.github.com/scott-codecov/311c174ecc7de87f7d7c50371c6ef927#file-cobertura-rs-L18-L31
-fn attributes_map(attributes: Attributes) -> HashMap<String, String> {
-    return attributes
-        .filter_map(|a| {
-            if let Ok(attr) = a {
-                let bytes = attr.value.into_owned();
-                let value = String::from_utf8(bytes).unwrap();
-                let key = String::from_utf8(attr.key.local_name().as_ref().to_vec()).unwrap();
-                Some((key, value))
-            } else {
-                None
-            }
-        })
-        .collect::<HashMap<_, _>>();
+fn attributes_map(attributes: Attributes) -> Result<HashMap<String, String>, pyo3::PyErr> {
+    let mut attr_map = HashMap::new();
+    for attribute in attributes {
+        if let Ok(attr) = attribute {
```

Comment:
Would this return a PyErr if this isn't true? Not super familiar w/ the syntax. Is it okay to not explicitly specify a PyErr?

Reply:
I don't think this will return an error, and we don't want it to, the goal is to just skip over `error` attributes